### PR TITLE
Upgrade to angular2 rc.5

### PR DESCRIPTION
**Pull request by DorianGrey**
This pull request should **not** be merged immediately, it's just a setup to better keep track what still has to be done, and what is already migrated.

After fighting jspm and Intellij for at least an hour (... rewritten configs without any kind of sense / mixing dependency links jspm_packages/node_modules, etc. ...), Had to revert the recent rxjs update due to version conflicts. I was able to bundle and start the application again. Not everything is working as expected (yet). The current state is mentioned below, and will be updated as soon as the topics are dealt with.
**Note**: I'm currently amending the changes to the first commit, so don't wonder why it's still marked with "x days ago"
**Note**: App is running again.
- [x] Updated the relevant dependencies
- [x] Adopted the entry points to use `@NgModule` style.
- [x] Drop ngrx/router in favor of angular/router, since it's no longer maintained.
- [x] Adopt module style for further development towards RC.6 and above.
- [x] Deal with the huge amount of warnings raised by the Typescript compiler, complaining about absent module definitions.
- [x] Fix the unit tests. These changes are not yet reflected in any official docs - most information is spread along the current docs and various changelog entries. Horrible. We might have a closer look at how the tests from https://github.com/angular/material2 are implemented - I'd suggest its better than waiting for the docs.
- [x] Fix hot reloading - it currently seems to run, but does not seem to do anything. Maybe it has to refer to the App component, and not to the application module. This issue might be related to https://github.com/capaj/systemjs-hot-reloader/issues/97. I've tracked it down to the point where the state reload attempt starts - with some minor adoptions, we're receiving a proper state. However, taking it over does not seem to work.
  **€dit**: It works fine again, but the required code still feels a little bit hacky ... :fearful:

### CODE REVIEW

**Comment by DorianGrey** [comment]
The list of typescripts errors is rather long atm. (n.b.: although this log is from Windows, the same occurs on Ubuntu, thus it's not specific to a particular platform) :

```
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/app.component.ts:3:25
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/ng2-translate@2.4.1/ng2-translate.d.ts:1:37
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/ng2-translate@2.4.1/src/translate.service.d.ts:1:30
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/ng2-translate@2.4.1/src/translate.service.d.ts:2:22
TypeScript [Error] Cannot find module '@angular/http'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/ng2-translate@2.4.1/src/translate.service.d.ts:3:28
TypeScript [Error] Cannot find module 'rxjs/Observable'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/ng2-translate@2.4.1/src/translate.pipe.d.ts:1:75
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/app.routes.ts:1:36
TypeScript [Error] Cannot find module '@angular/router'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/inputTest/inputTest.component.ts:1:25
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/inputTest/inputTest.routes.ts:1:28
TypeScript [Error] Cannot find module '@angular/router'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/inputTest/input.module.ts:10:24
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/inputTest/input.module.ts:11:29
TypeScript [Error] Cannot find module '@angular/platform-browser'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@angular/forms@0.3.0/src/directives.d.ts:8:22
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@angular/forms@0.3.0/src/directives/abstract_form_group_directive.d.ts:8:35
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@angular/forms@0.3.0/src/directives/checkbox_value_accessor.d.ts:8:38
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@angular/forms@0.3.0/src/directives/default_value_accessor.d.ts:8:38
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@angular/forms@0.3.0/src/directives/control_value_accessor.d.ts:8:29
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@angular/forms@0.3.0/src/directives/ng_model.d.ts:8:53
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@angular/forms@0.3.0/src/directives/ng_model_group.d.ts:8:35
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@angular/forms@0.3.0/src/directives/reactive_directives/form_control_directive.d.ts:8:42
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@ngrx/store@2.1.2/reducer.d.ts:1:33
TypeScript [Error] Cannot find module 'rxjs/BehaviorSubject'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@ngrx/store@2.1.2/dispatcher.d.ts:1:33
TypeScript [Error] Cannot find module 'rxjs/BehaviorSubject'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@ngrx/store@2.1.2/state.d.ts:1:33
TypeScript [Error] Cannot find module 'rxjs/BehaviorSubject'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@ngrx/store@2.1.2/store.d.ts:2:26
TypeScript [Error] Cannot find module 'rxjs/Observer'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@ngrx/store@2.1.2/store.d.ts:3:28
TypeScript [Error] Cannot find module 'rxjs/Observable'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@ngrx/store@2.1.2/store.d.ts:4:26
TypeScript [Error] Cannot find module 'rxjs/Operator'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@ngrx/store@2.1.2/ng2.d.ts:5:50
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/jspm_packages/npm/@ngrx/core@1.0.2/operator/select.d.ts:1:28
TypeScript [Error] Cannot find module 'rxjs/Observable'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todo.service.spec.ts:4:8
TypeScript [Error] Cannot find module '@angular/core/testing'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todo.service.spec.ts:12:26
TypeScript [Error] Cannot find module 'rxjs/Observable'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todo.service.ts:4:26
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todo.service.ts:5:26
TypeScript [Error] Cannot find module 'rxjs/Observable'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todos.component.ts:1:36
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todos.component.ts:3:26
TypeScript [Error] Cannot find module 'rxjs/Observable'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todos.component.ts:25:3
TypeScript [Error] Member 'name' implicitly has an 'any' type. (TS7008)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todos.component.spec.ts:1:25
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todos.component.spec.ts:2:18
TypeScript [Error] Cannot find module '@angular/platform-browser'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todos.component.spec.ts:4:38
TypeScript [Error] Cannot find module '@angular/core/testing'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todos.module.ts:1:24
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todos.module.ts:2:29
TypeScript [Error] Cannot find module '@angular/platform-browser'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/app/todos/todos.routes.ts:2:28
TypeScript [Error] Cannot find module '@angular/router'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/main.dev.ts:4:38
TypeScript [Error] Cannot find module '@angular/platform-browser-dynamic'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/main.dev.ts:5:24
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/main.dev.ts:6:26
TypeScript [Error] Cannot find module '@angular/http'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/main.dev.ts:44:23
TypeScript [Error] Property 'take' does not exist on type 'Store<any>'. (TS2339)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/main.dev.ts:44:41
TypeScript [Error] Parameter 's' implicitly has an 'any' type. (TS7006)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/main.dev.ts:48:11
TypeScript [Error] Parameter 'ref' implicitly has an 'any' type. (TS7006)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/main.ts:4:38
TypeScript [Error] Cannot find module '@angular/platform-browser-dynamic'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/main.ts:5:40
TypeScript [Error] Cannot find module '@angular/core'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/main.ts:6:26
TypeScript [Error] Cannot find module '@angular/http'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/test-setup.ts:12:23
TypeScript [Error] Cannot find module '@angular/core/testing'. (TS2307)
TypeScript [Error] file:///D:/GitHub/ng2-jspm-template/src/test-setup.ts:16:8
TypeScript [Error] Cannot find module '@angular/platform-browser-dynamic/testing'. (TS2307)
```

@svi3c - any ideas? All kind of typings, regardless of whether they are mentioned in jspm.config.js or referenced via path in tsconfig.json, seem to be ignored, which doesn't make sense (it does not even break the build) - things work fine when running the app.

File: src/main.dev.ts [hunk 1]
```diff
@@ -1,39 +1,59 @@
...
-export  let __reload = (m: {storeHolder: typeof storeHolder}) =>
-  m.storeHolder.store.take(1).subscribe(s => oldState = s);
+export let __reload = (m: {storeHolder: typeof storeHolder}) => {
+  m.storeHolder.store.take(1).subscribe(toBeReloadedState => {
+    // toBeReloadedState => Entries from store from <before> the reload.
+    // targetState       => State to be used to initialize the store.
+    // Note: It seems that the initialization happens BEFORE this subscription is triggered for the first time,
+    // i.e. targetState need to be defined in advance, and the former state has to be assigned
+    // to the same object to properly get adopted to all components.
+    assign(targetState, toBeReloadedState);
```
Comment (DorianGrey) [inline src/main.dev.ts:51]: @svi3c : Any better ideas how to solve this? 
I'm assigning all parts of the initial state to `targetState` within `createStoreProvider` (in `app.store.ts`), which seems to always happen **before** we're landing here. Thus, to properly adopt the former state, the `targetState`object has to be manipulated after this step happens, Although it works, this still feels a little bit ... hacky.

Comment (sv2dev) [inline src/main.dev.ts:51]: It is the way to go here atm. I'll create a PR which will improve this part as well.

**Comment by sv2dev** [comment]
@DorianGrey the typescript errors can be resolved when you add the typings meta flag for the libs.
You just have to run `jspm install` again afterwards. I just added them and it works for me.

**Pull request merged**